cmake/GenerateVersion.cmake

```diff
@@ -19,30 +13,29 @@ execute_process(
   RESULT_VARIABLE RES
 )
 
-if("${RES}" EQUAL 1)
+if(NOT RES EQUAL 0)
+  message(STATUS "Git tag extraction failed with \n"  "   ${ERR}" )
   if(EXISTS ${OUTPUT})
     message(STATUS "Unable to extract version-string from git: keeping the last known version")
```

Comment:
This is again a case of being clever for no reason. 

1. Using the old version is misleading.
2. The extra logic here adds noise to the code.